fem/bilininteg.hpp

```diff
@@ -2889,6 +2918,62 @@ class VectorInnerProductInterpolator : public DiscreteInterpolator
    VectorCoefficient *VQ;
 };
 
+// Local maximum size of dofs and quads in 1D
+constexpr int HDIV_MAX_D1D = 5;
+constexpr int HDIV_MAX_Q1D = 6;
+
+void PAHdivMassApply3D(const int D1D,
+                       const int Q1D,
+                       const int NE,
+                       const Array<double> &_Bo,
+                       const Array<double> &_Bc,
+                       const Array<double> &_Bot,
+                       const Array<double> &_Bct,
+                       const Vector &_op,
+                       const Vector &_x,
+                       Vector &_y);
+
+void PAHdivMassApply2D(const int D1D,
+                       const int Q1D,
+                       const int NE,
+                       const Array<double> &_Bo,
+                       const Array<double> &_Bc,
+                       const Array<double> &_Bot,
+                       const Array<double> &_Bct,
+                       const Vector &_op,
+                       const Vector &_x,
+                       Vector &_y);
+
+void PAHdivSetup3D(const int Q1D,
+                   const int NE,
+                   const Array<double> &w,
+                   const Vector &j,
+                   Vector &_coeff,
+                   Vector &op);
+
+void PAHdivSetup2D(const int Q1D,
+                   const int NE,
+                   const Array<double> &w,
+                   const Vector &j,
+                   Vector &_coeff,
+                   Vector &op);
+
+void PAHdivMassAssembleDiagonal3D(const int D1D,
+                                  const int Q1D,
+                                  const int NE,
+                                  const Array<double> &_Bo,
+                                  const Array<double> &_Bc,
+                                  const Vector &_op,
+                                  Vector &_diag);
+
+void PAHdivMassAssembleDiagonal2D(const int D1D,
+                                  const int Q1D,
+                                  const int NE,
+                                  const Array<double> &_Bo,
+                                  const Array<double> &_Bc,
+                                  const Vector &_op,
```

Comment:
It looks like these functions are here because they are defined in one `*.cpp` file and then used in another `*.cpp` file. I think we can move these declarations to the file where they are used -- they do not need to be visible by when including `mfem.hpp`.

The `constexpr` constants `HDIV_MAX_D1D` and `HDIV_MAX_Q1D` can be moved to the implementation file -- they are probably used just there.